src/libsyntax/ext/expand.rs

```diff
@@ -400,6 +400,27 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
         result
     }
 
+    fn fully_configure(&mut self, item: Annotatable) -> Annotatable {
+        let mut cfg = StripUnconfigured {
+            should_test: self.cx.ecfg.should_test,
+            sess: self.cx.parse_sess,
+            features: self.cx.ecfg.features,
+        };
+        // Since the item itself has already been configured by the InvocationCollector,
+        // we know that fold result vector will contain exactly one element
+        match item {
+            Annotatable::Item(item) => {
+                Annotatable::Item(cfg.fold_item(item).pop().unwrap())
+            }
+            Annotatable::TraitItem(item) => {
+                Annotatable::TraitItem(P(cfg.fold_trait_item(item.unwrap()).pop().unwrap()))
```

Comment:
nit: `item.map(|item| cfg.fold_trait_item(item).pop().unwrap())` is more idiomatic and efficient here (i.e. with `P`). Same for `ImplItem`.

Reply:
Got it, fixed.